pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UseEnumCollectionsRule.java

```diff
@@ -33,8 +35,11 @@ public Object visit(ASTConstructorCall call, Object data) {
         JTypeMirror builtType = call.getTypeMirror();
 
         if (!builtType.isRaw()) {
-            boolean isMap = TypeTestUtil.isExactlyA(HashMap.class, builtType);
-            if (isMap || TypeTestUtil.isExactlyA(HashSet.class, builtType)) {
+            boolean isMap = TypeTestUtil.isExactlyA(HashMap.class, builtType)
+                || TypeTestUtil.isExactlyA(TreeMap.class, builtType);
+            if (isMap
+                || TypeTestUtil.isExactlyA(HashSet.class, builtType)
+                || TypeTestUtil.isExactlyA(TreeSet.class, builtType)) {
```

Comment:
I'm not sure we want to flag `TreeSet` / `TreeMap`, as those are navigable, with a well known iteration order, which `EnumSet` / `EnumMap` do not provide…

Reply:
Enummap and enumset do specify an iteration order (natural order), although they don't implement SortedMap/Set. I think maybe we should not flag them, as the "default" implementation of a map is anyway HashMap, so we probably won't catch many treemaps anyway.